homeassistant/components/sonarr/calendar.py

```diff
@@ -0,0 +1,93 @@
+"""Support for Sonarr calendar."""
+from __future__ import annotations
+
+from datetime import datetime, timedelta
+
+from homeassistant.components.calendar import CalendarEntity, CalendarEvent
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity import EntityDescription
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util.dt import as_utc, start_of_local_day
+
+from .const import DOMAIN
+from .coordinator import CalendarDataUpdateCoordinator
+from .entity import SonarrEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Sonarr calendar based on a config entry."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]["upcoming"]
+    async_add_entities([SonarrCalendarEntity(coordinator, "Episodes")])
+
+
+class SonarrCalendarEntity(SonarrEntity, CalendarEntity):
+    """Defines a Sonarr Calendar."""
```

Comment:
Since the default update interval of the calendar coordinator is 30 seconds and the default update interval using the builtin polling of the calendar entity is 60 seconds, I think we can keep the builtin polling disabled, as it normally is for coordinator entities. We won't lose state updates between coordinator refreshes.